OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -330,6 +330,42 @@ public void SetLineWidth(float width)
 			ErrorHandler.CheckGlError();
 		}
 
+		public Bitmap TakeScreenshot()
+		{
+			var rect = new Rectangle(Point.Empty, size);
+			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var data = bitmap.LockBits(rect,
+				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+
+			try
+			{
+				int packAlignment, unpackAlignment;
+
+				unsafe
+				{
+					GL.GetInteger(GetPName.PackAlignment, out packAlignment);
+					GL.GetInteger(GetPName.UnpackAlignment, out unpackAlignment);
+				}
+
+				GL.PixelStore(PixelStoreParameter.UnpackAlignment, 1);
+				GL.PixelStore(PixelStoreParameter.PackAlignment, 1);
+
+				GL.ReadPixels(rect.X, rect.Y, rect.Width, rect.Height, PixelFormat.Bgra, PixelType.UnsignedByte, data.Scan0);
+				GL.Finish();
+
+				GL.PixelStore(PixelStoreParameter.UnpackAlignment, unpackAlignment);
```

Comment:
> You can just safely leave the alignment at 1.

I was told this is bad practice, because you never know if someone else needs it in a different value for a future engine feature so I took the extra effort to avoid magic breakage.

> Is it necessary to use `UnpackAlignment` at all? Since the frame buffer is only being read from and not written to...

Removed.